packages/design-system/src/components/IdleTimeout/IdleTimeout.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable react/no-multi-comp */
+/* eslint-disable react/no-danger */
+import React from 'react';
+import { Dialog } from '../Dialog';
+import { Button } from '../Button';
+
+export interface IdleTimeoutProps {
+  /**
+   * The text for the 'close' button on the warning dialog
+   */
+  closeDialogText?: string;
+  /**
+   * The text for the 'continue session' button in warning dialog.
+   */
+  continueSessionText?: string;
+  /**
+   * The heading text for the warning dialog.
+   */
+  heading?: string;
+  /**
+   * The text for the button that ends the session in warning dialog.
+   */
+  endSessionButtonText?: string;
+  /**
+   * The message text for the warning dialog.
+   * Note that using the token `<timeToTimeout>` will be replaced in the message text with the number of minutes until timeout.
+   */
+  message?: string;
+  /**
+   * Optional function that is called when the warning dialog's close button is clicked
+   */
+  onClose?: (...args: any[]) => any;
+  /**
+   * Optional function that is called when the user chooses to keep the session alive.
+   * The IdleTimeout component will reset the countdown internally.
+   */
+  onSessionContinue?: (...args: any[]) => any;
+  /**
+   * Optional function that is called when the session is manually ended by user.
+   * If not provided, the behavior of `onTimeout` will be used.
+   */
+  onSessionForcedEnd?: (...args: any[]) => any;
+  /**
+   * Function that is called when the timeout countdown reaches zero.
+   */
+  onTimeout: (...args: any[]) => any;
+  /**
+   * Describes if the button to manually end session should be shown
+   */
+  showSessionEndButton?: boolean;
+  /**
+   * Defines the time (in minutes) until the session is timed out
+   */
+  timeToTimeout: number;
+  /**
+   * Defines the time (in minutes) until the warning message is shown. The default is 5 minutes.
+   */
+  timeToWarning?: number;
+}
+
+const IdleTimeout = ({
+  closeDialogText = 'Close',
+  continueSessionText = 'Continue session',
+  heading = 'Are you still there?',
+  endSessionButtonText = 'Logout',
+  message = 'You’ve been inactive for a while. <br/>Your session will end in <timeToTimeout>. <br/><br/>Select "Continue session" below if you want more time.',
```

Comment:
This is clever, but I think our DS has styles (spacing) associated with true paragraphs that aren't available when `<br>` tags are used instead. To your comment above, you might need to make this a React.Node type to ensure our styles are applied to whatever message is passed - or we might need to add more robust styles on our end to account for using `<br>` tags